avocado/core/runner.py

```diff
@@ -239,7 +239,8 @@ def finish(self, proc, started, timeout, step):
                        test_state['name'])
         if proc.is_alive():
             TEST_LOG.warning("Killing hanged test process %s" % proc.pid)
-            for _ in xrange(5):     # I really want to destroy it
+            end_time = time.time() + 60
+            while time.time() < end_time:
```

Comment:
Wow this is way more aggressive. We are going from 5 `os.kill()` calls in 0.5s to ~600 calls in 60s. Do we need that much of calls or all we need is to wait for one call to have effect?